qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -103,20 +103,26 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
             The experiment circuits
 
         Raises:
-            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration
+            AttributeError: if unit is 'dt', but 'dt' parameter is missing in the backend configuration.
+            ValueError: if values in `delays` are not strictly increasing.
         """
         if self._unit == "dt":
             try:
                 dt_factor = getattr(backend._configuration, "dt")
             except AttributeError as no_dt:
                 raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+        if not all(d1 < d2 for d1, d2 in zip(self._delays, self._delays[1:])):
```

Comment:
It shouldn't matter what order delays are in and circuits are run if they delay time is stored in circuit metadata and we make the min value check more robust as suggested below